src/libs/TransactionUtils.ts

```diff
@@ -581,8 +581,8 @@ function getAllReportTransactions(reportID?: string, transactions?: OnyxCollecti
     // `reportID` from the `/CreateDistanceRequest` endpoint return's number instead of string for created `transaction`.
     // For reference, https://github.com/Expensify/App/pull/26536#issuecomment-1703573277.
     // We will update this in a follow-up Issue. According to this comment: https://github.com/Expensify/App/pull/26536#issuecomment-1703591019.
-    const nonNullableTransactions: Transaction[] = Object.values(transactions ?? allTransactions ?? {}).filter((transaction): transaction is Transaction => transaction !== null);
-    return nonNullableTransactions.filter((transaction) => `${transaction.reportID}` === `${reportID}`);
+    const nonNullableTransactions = Object.values(transactions ?? allTransactions ?? {}).filter((transaction) => !!transaction);
+    return nonNullableTransactions.filter((transaction) => `${transaction?.reportID}` === `${reportID}`);
```

Comment:
Looks like the ts check is a bit weird now. The CI ts check complains that the value is possibly undefined even though it's not locally.

_I can't even do .filter(Boolean)_

Reply:
Hmm, the CI still complains

Reply:
Done